core/src/main/java/io/questdb/std/IntStack.java

```diff
@@ -69,6 +69,19 @@ public int pop() {
         return result;
     }
 
+    public int pollLast() {
```

Comment:
This method seems to break the contract of the class in a number of ways - it breaks the "stack is empty when head == tail" contract, as well as doesn't check for `noEntryValue` and doesn't replace the removed element with `noEntryValue`:
```java
    @Test
    public void testPollLast() {
        int sz = 2050;
        int[] expected = new int[sz];
        int l = expected.length - 1;

        IntStack stack = new IntStack();
        for (int i = expected.length - 1; i > -1; i--) {
            stack.push(expected[l--] = i);
        }

        Assert.assertEquals(sz, stack.size());

        int count = expected.length - 1;
        while (stack.notEmpty()) {
            Assert.assertEquals("at " + count, expected[count--], stack.pollLast());
        }
        // The following assertions fail while they shouldn't.
        Assert.assertFalse(stack.notEmpty());
        Assert.assertEquals(-1, stack.pollLast());
    }
```

Reply:
thank you Andrey, very good spot. Fixed here: https://github.com/questdb/questdb/pull/2037/commits/c5a5a83266c0d88716b16b5782fb6b9718aeeda4